pkg/controller/spinnakervalidating/tls.go

```diff
@@ -0,0 +1,195 @@
+package spinnakervalidating
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	"math"
+	"math/big"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+const (
+	certName = "tls.crt"
+	keyName  = "tls.key"
+	caName   = "ca.crt"
+
+	certificateBlockType = "CERTIFICATE"
+	rsaKeySize           = 2048
+	duration365d         = time.Hour * 24 * 365
+)
+
+type certContext struct {
+	cert        []byte
+	key         []byte
+	signingCert []byte
+	certDir     string
+}
+
+func getCertContext(operatorNamespace string, operatorServiceName string) (*certContext, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+	certDir := filepath.Join(home, "spinnaker-operator-certs")
```

Comment:
It seems more standard to take `--tls-key` --tls-cert` parameters (that default to `/etc/certs/tls.key|crt`. If the files are not found they are auto-generated, otherwise people can mount their own cert to these locations and potentially override them via the parameters in the deployment.

Reply:
We can make configurable `certDir`, but the sdk has these restrictions:
* Cert and Key files must be on the same dir
* Cert and Key file names must be named `tls.crt` and `tls.key`, these names are hardcoded in the sdk